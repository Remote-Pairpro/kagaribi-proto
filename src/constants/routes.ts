export const HOME = "/";
export const ABOUT = "/about";
export const SIGN_UP = "/signup";
export const SIGN_IN = "/signin";
