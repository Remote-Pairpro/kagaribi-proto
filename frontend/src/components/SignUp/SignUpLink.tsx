import React from "react";
import { Link } from "react-router-dom";
import * as routes from "../../constants/routes";

const SignUpLink = () => {
    return (
        <p>
            アカウントをお持ちでないですか？{" "}
            <Link to={routes.SIGN_UP}>Sign Up</Link>
        </p>
    );
};

export default SignUpLink;
