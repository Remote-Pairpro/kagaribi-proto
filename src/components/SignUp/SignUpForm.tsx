import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import React from "react";
import * as routes from "../../constants/routes";
import * as auth from "../../firebase/auth";

interface ISignUpFormProps {
    history: any;
}
interface ISignUpFormState {
    username: string;
    email: string;
    passwordOne: string;
    passwordTwo: string;
    error: any;
}

const INITIAL_STATE: ISignUpFormState = {
    username: "",
    email: "",
    passwordOne: "",
    passwordTwo: "",
    error: null,
};

class SignUpForm extends React.Component<ISignUpFormProps, ISignUpFormState> {
    constructor(signUpFormProps: ISignUpFormProps) {
        super(signUpFormProps);
        this.state = { ...INITIAL_STATE };
    }

    public onSubmit = (event: any): void => {
        const signUpFormState: ISignUpFormState = this.state;
        const signUpFormProps: ISignUpFormProps = this.props;

        auth.signUpWithEamilAndPassword(
            signUpFormState.email,
            signUpFormState.passwordOne
        )
            .then(authUser => {
                this.setState({ ...INITIAL_STATE });
                signUpFormProps.history.push(routes.HOME);
            })
            .catch(error => {
                this.setState({ error });
            });

        event.preventDefault();
    };

    public render() {
        const states: ISignUpFormState = this.state;
        const isInvalid: boolean =
            states.passwordOne !== states.passwordTwo ||
            states.passwordOne === "" ||
            states.email === "" ||
            states.username === "";
        return (
            <form onSubmit={this.onSubmit}>
                <TextField
                    id="Full Name"
                    label="Full Name"
                    type="text"
                    value={states.username}
                    onChange={event =>
                        this.setState({ username: event.target.value })
                    }
                    placeholder="Full Name"
                    autoComplete="on"
                />
                {"  "}
                <TextField
                    id="email-address"
                    label="Email"
                    type="text"
                    value={states.email}
                    onChange={event =>
                        this.setState({ email: event.target.value })
                    }
                    placeholder="Email Address"
                    autoComplete="on"
                />
                {"  "}
                <TextField
                    id="password"
                    label="Password"
                    type="password"
                    value={states.passwordOne}
                    onChange={event =>
                        this.setState({ passwordOne: event.target.value })
                    }
                    placeholder="Password"
                    autoComplete="on"
                />
                {"  "}
                <TextField
                    id="password-two"
                    label="Confirm Password"
                    type="password"
                    value={states.passwordTwo}
                    onChange={event =>
                        this.setState({ passwordTwo: event.target.value })
                    }
                    placeholder="Confirm Password"
                    autoComplete="on"
                />
                <Button type="submit" disabled={isInvalid} variant="outlined">
                    Sign Up
                </Button>
                {states.error && <p>{states.error.message}</p>}
            </form>
        );
    }
}

export default SignUpForm;
export { ISignUpFormProps };
