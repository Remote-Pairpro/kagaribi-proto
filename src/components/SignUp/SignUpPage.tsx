import React from "react";
import { withRouter } from "react-router-dom";
import SignUpForm, { ISignUpFormProps } from "./SignUpForm";

const SignUpPage = (signUpFormProps: ISignUpFormProps) => {
    return (
        <div>
            <h1>SignUp</h1>
            <SignUpForm history={signUpFormProps.history} />
        </div>
    );
};

export default withRouter(SignUpPage);
