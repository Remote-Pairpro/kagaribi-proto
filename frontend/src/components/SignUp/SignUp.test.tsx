import { render, shallow } from "enzyme";
import { createMemoryHistory } from "history";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import * as routes from "../../constants/routes";
import SignUpForm from "./SignUpForm";
import SignUpLink from "./SignUpLink";
import SignUpPage from "./SignUpPage";

describe("<SignUpLink />", () => {
    it("should have only one link", () => {
        const wrapper = render(
            <MemoryRouter initialEntries={[routes.SIGN_UP]}>
                <SignUpLink />
            </MemoryRouter>
        );
        expect(wrapper.find("a").length).toBe(1);
    });

    it("should be rendered correctly", () => {
        const tree = renderer
            .create(
                <MemoryRouter initialEntries={[routes.SIGN_UP]}>
                    <SignUpLink />
                </MemoryRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<SignUpForm />", () => {
    it("should be rendered correctly", () => {
        const history = createMemoryHistory();
        const tree = shallow(<SignUpForm history={history} />, {
            disableLifecycleMethods: true,
        });
        expect(tree).toMatchSnapshot();
    });
});

describe("<SignUpPage />", () => {
    it("should be rendered correctly", () => {
        const tree = shallow(<SignUpPage />);
        expect(tree).toMatchSnapshot();
    });
});
