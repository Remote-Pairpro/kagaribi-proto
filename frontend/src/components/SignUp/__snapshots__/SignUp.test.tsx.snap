// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpForm /> should be rendered correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <TextField
    autoComplete="on"
    id="Full Name"
    label="Full Name"
    onChange={[Function]}
    placeholder="Full Name"
    required={false}
    select={false}
    type="text"
    value=""
    variant="standard"
  />
    
  <TextField
    autoComplete="on"
    id="email-address"
    label="Email"
    onChange={[Function]}
    placeholder="Email Address"
    required={false}
    select={false}
    type="text"
    value=""
    variant="standard"
  />
    
  <TextField
    autoComplete="on"
    id="password"
    label="Password"
    onChange={[Function]}
    placeholder="Password"
    required={false}
    select={false}
    type="password"
    value=""
    variant="standard"
  />
    
  <TextField
    autoComplete="on"
    id="password-two"
    label="Confirm Password"
    onChange={[Function]}
    placeholder="Confirm Password"
    required={false}
    select={false}
    type="password"
    value=""
    variant="standard"
  />
  <WithStyles(Button)
    disabled={true}
    type="submit"
    variant="outlined"
  >
    Sign Up
  </WithStyles(Button)>
</form>
`;

exports[`<SignUpLink /> should be rendered correctly 1`] = `
<p>
  アカウントをお持ちでないですか？
   
  <a
    href="/signup"
    onClick={[Function]}
  >
    Sign Up
  </a>
</p>
`;

exports[`<SignUpPage /> should be rendered correctly 1`] = `
<Route>
  <Component />
</Route>
`;
